unittests/forked_tests_if.cpp

```diff
@@ -153,17 +188,530 @@ BOOST_FIXTURE_TEST_CASE(fork_with_bad_block_if, savanna_cluster::cluster_t) try
+   }
+
+} FC_LOG_AND_RETHROW()
+
+// ------------------------------ split_and_rejoin ---------------------------------------
+// demonstrates a worst-case scenario of finalizers split apart for different branches,
+// then rejoin together, and need to reach consensus on one of the branches through their
+// votes.
+// ---------------------------------------------------------------------------------------
+BOOST_FIXTURE_TEST_CASE( split_and_rejoin, savanna_cluster::cluster_t ) try {
+   const vector<account_name> producers { "p1"_n, "p2"_n, "p3"_n };
+   _nodes[0].create_accounts(producers);
+   set_producers(0, producers);               // set new producers and produce blocks until the switch is pending
+   _nodes[0].create_accounts( {"alice"_n} );
+   _nodes[0].produce_blocks(12);
+   auto lib0 = _nodes[0].control->last_irreversible_block_num();
+   wlog("lib0 = ${lib0}", ("lib0", lib0)); // 45
+
+   // split the network
+   const std::vector<size_t> partition {2, 3};
+   set_partition(partition);       // simulate 2 disconnected partitions:  nodes {0, 1} and node {2, 3}
+
+   // produce 12 blocks on _nodes[0]'s partition
+   _nodes[0].create_accounts( {"bob"_n} );
+   _nodes[0].produce_blocks(12);
+   BOOST_CHECK_EQUAL( _nodes[0].control->last_irreversible_block_num(), lib0 + 1);
+   BOOST_CHECK_EQUAL( does_account_exist( _nodes[0], "alice"_n ), true );
+   BOOST_CHECK_EQUAL( does_account_exist( _nodes[0], "bob"_n ),   true );
+
+   // produce 12 blocks on _nodes[2]'s partition
+   _nodes[2].produce_block(_block_interval_us * 13);
+   _nodes[2].create_accounts( {"carol"_n} );
+   _nodes[2].produce_blocks(11);
+   BOOST_CHECK_EQUAL( _nodes[2].control->last_irreversible_block_num(), lib0 + 1);
+   BOOST_CHECK_EQUAL( does_account_exist( _nodes[2], "alice"_n ), true );
+   BOOST_CHECK_EQUAL( does_account_exist( _nodes[2], "bob"_n ),   false );
+   BOOST_CHECK_EQUAL( does_account_exist( _nodes[2], "carol"_n ), true );
+
+   // update the network split so that {0, 1, 2} are in one partition, enough for finality to start
+   // advancing again
+   set_partition(std::vector<size_t>{3});       // simulate 2 disconnected partitions:  nodes {0, 1, 2} and node {3}
+
+   propagate_heads();                           // otherwise we get unlinkable_block when newly produced blocks are pushed to node2
+
+   // and restart producing on _nodes[0]
+   _nodes[0].produce_block(_block_interval_us * 26, true);
+   _nodes[0].produce_blocks(6);
+   auto lib2 = _nodes[0].control->last_irreversible_block_num();
+   BOOST_CHECK_EQUAL(lib2, lib0 + 12 + 7 );   // 12 when network was split, 7 just above (6 + 1)
+   wlog("lib2 = ${lib2}", ("lib2", lib2)); // 65
+
+} FC_LOG_AND_RETHROW()
+
+// ---------------------------- push_block_returns_forked_transactions_if ---------------------------------
+BOOST_FIXTURE_TEST_CASE( push_block_returns_forked_transactions_if, savanna_cluster::cluster_t  ) try {
```

Comment:
I think we should remove `if` or change to `savanna`

Reply:
done